MachineLearningPrototype/collect_data.py

```diff
@@ -0,0 +1,132 @@
+import requests
+import argparse
+from datetime import datetime
+from datetime import date
+from math import ceil
+import time
+from pprint import pprint
+
+def get_num_pages(repo):
+	"""
+	Fetch the number of pages that store issues
+	"""
+	repo_query_url = f'https://api.github.com/repos/{repo}'
+	num_issues = requests.get(repo_query_url).json()['open_issues_count']
+	num_pages = ceil(num_issues/100)
+	return num_pages
+
+def fetch_all_new_issues(repo, since, auth_token, labels):
+	"""
+	Fetch all new open issues in the repo
+	"""
+	num_pages = get_num_pages(repo)
+	pprint(f'Need to query {num_pages} pages')
+
+	page_number = 1
+
+	open_issues = []
+	
+	query_url = f'https://api.github.com/repos/{repo}/issues'
+	params = {'accept': 'application/vnd.github.v3+json',
+			  'state': 'open',
+			  'since': since,
+			  'per_page': 100,
+			  'page': page_number,}
+
+	headers = None
+	if auth_token:
+		headers = {'Authorization': f'token {auth_token}'}
+
+	while page_number <= num_pages:
+		params['page'] = page_number
+		response = requests.get(query_url, headers=headers, params=params).json()
+		open_issues.extend([r for r in response if 'pull_request' not in r])
+		pprint(f'Found {len(open_issues)} open issues')
+		page_number += 1
+	
+	pprint(f'Found {len(open_issues)} open issues in total')
+	since = datetime.now().isoformat()
+	
+	return since
+
+def fetch_issues_with_label(query_url, headers, params, label):
+	"""
+	Fetch all new open issues in the repo with a given label
+	"""
+	open_issues = []
+	page_number = 1
+	params['labels'] = label
+		
+	while True:
+		params['page'] = page_number
+		response = requests.get(query_url, headers=headers, params=params).json()
+		open_issues_on_page = [r for r in response if 'pull_request' not in r]
+		
+		if len(open_issues_on_page) == 0:
+			break
+		
+		open_issues.extend(open_issues_on_page)
+		pprint(f'Found {len(open_issues_on_page)} open issues')
+		page_number += 1	
+
+	pprint(f'Found {len(open_issues)} open issues in total')
+	
+def fetch_issues_with_labels(repo, since, auth_token, labels):
+	"""
+	Fetch all new open issues in the repo with the given labels
+	"""
+	query_url = f'https://api.github.com/repos/{repo}/issues'
+	params = {'accept': 'application/vnd.github.v3+json',
+			  'state': 'open',
+			  'since': since,
+			  'per_page': 100,
+			  'page': 1,
+			  'labels': None,}
+
+	headers = None
+	if auth_token:
+		headers = {'Authorization': f'token {auth_token}'}
+
+	for label in labels:
+		fetch_issues_with_label(query_url, headers, params, label)
+
+	return datetime.now().isoformat()
```

Comment:
Did this return value get used?

Reply:
Yes it is used to update the since parameter. Sorry forgot to add it to the main function.